src/main/java/com/volmit/iris/engine/actuator/IrisDecorantActuator.java

```diff
@@ -57,6 +61,21 @@ public IrisDecorantActuator(Engine engine) {
         seaSurfaceDecorator = new IrisSeaSurfaceDecorator(getEngine());
         shoreLineDecorator = new IrisShoreLineDecorator(getEngine());
         seaFloorDecorator = new IrisSeaFloorDecorator(getEngine());
+
```

Comment:
This is being redefined every time a new DecorantActuator is created. Set this value on line 41 where the field is defined. If you cant do this because you are using local variables in it, then it shouldnt be a static field.

Reply:
Additionally with multi-engine environment or multi-iris-world environment this will fuck shit up because of the above.

Reply:
I originally had it there but because it has calls to getEngine() which isn't static, it had a hissy fit. I'll have a think about this

Reply:
Then dont make PREDICATE_CAVELIQUID static, because now, it will only initialize on the first engine it sees. Every new studio world will throw errors, and cause issues if the engine changes. Just set predicateLiquid to NOT static and problem solved